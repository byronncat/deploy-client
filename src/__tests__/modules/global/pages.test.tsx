import {
  BrowserRouter,
  RouterProvider,
  createBrowserRouter,
} from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';

import { ErrorPage } from '@global';

describe('pages', () => {
  it('should render the 404 page correctly', () => {
    const { container } = render(
      <BrowserRouter>
        <ErrorPage />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  test('can navigate to home page', () => {
    window.history.pushState({}, '', '/bad/route');

    const router = createBrowserRouter([
      {
        path: '*',
        element: <ErrorPage />,
      },
      {
        errorElement: <ErrorPage />,
        children: [{ path: '/', element: <div>Homepage</div> }],
      },
    ]);

    const { getByText } = render(<RouterProvider router={router} />);

    const button = getByText(/back to home/i);
    fireEvent.click(button);
    expect(window.location.pathname).toBe('/');
  });
});
