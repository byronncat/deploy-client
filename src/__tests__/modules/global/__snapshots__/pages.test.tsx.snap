// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages should render the 404 page correctly 1`] = `
<div>
  <div
    class="h-screen bg-haiti-950 flex flex-col md:flex-row overflow-hidden"
  >
    <div
      class="w-full md:w-1/2 h-1/2 md:h-full flex place-content-center"
    >
      <span
        class="square flicker-slightly w-36 md:w-1/2 w-36 md:h-full relative -rotate-24 flex justify-center items-center after:content-[\\"\\"] after:block after:w-full after:pb-full after:absolute after:rounded-2xl"
      >
        <p
          class="relative z-10 font-bold text-10vw text-haiti-950 drop-shadow-xl select-none"
        >
          404
        </p>
      </span>
    </div>
    <div
      class="w-full md:w-1/2 p-8 md:p-4 relative z-20 flex place-content-center flex-col"
    >
      <h4
        class="font-semibold text-lg"
      >
        Oops! Page not found.
      </h4>
      <p
        class="pt-3 pb-5"
      >
        The page you are looking for does not exist. Go back to the main page.
      </p>
      <a
        class="button self-center md:self-start"
        href="/"
      >
        Back to Home
      </a>
    </div>
  </div>
</div>
`;
